dali/pipeline/operators/crop/bbox_crop.h

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_CROP_BBOX_CROP_H_
+#define DALI_PIPELINE_OPERATORS_CROP_BBOX_CROP_H_
+
+#include <vector>
+#include <utility>
+#include <algorithm>
+
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+
+namespace dali {
+
+class BBoxCrop : public Operator<CPUBackend> {
+  static const unsigned int kAttempts = 100;
+  static const unsigned int kBboxSize = 4;
+
+ protected:
+  struct Bounds {
+    explicit Bounds(std::vector<float> &&bounds)
+        : min(bounds[0]), max(bounds[1]) {
+      DALI_ENFORCE(min >= 0, "Min should be at least 0.0. Received: " +
+                                 std::to_string(min));
+      DALI_ENFORCE(bounds.size() == 2, "Bounds should be provided as 2 values");
```

Comment:
If it is possible for `bounds.size() == 1` then max(bounds[1]) in intializer list is UB.